pgrx-sql-entity-graph/src/used_type.rs

```diff
@@ -342,9 +342,8 @@ impl UsedType {
                 /// Set via the type being an `Option`.
                 optional: #optional,
                 metadata: {
-                    use ::pgrx::pgrx_sql_entity_graph::metadata::PhantomDataExt;
-                    let marker: core::marker::PhantomData<#resolved_ty> = core::marker::PhantomData;
-                    marker.entity()
+                    use ::pgrx::pgrx_sql_entity_graph::metadata::SqlTranslatable;
+                    <#resolved_ty>::entity()
```

Comment:
Since this seems to take care of both use sites for PhantomDataExt, we can simply delete that trait too, in pgrx-sql-entity-graph/src/metadata/phantomdata_ext.rs and make this a nice -100loc PR.